src/Compilers/VisualBasic/Portable/LanguageFeatures/LanguageFeatureRequiredVersionService.vb

```diff
@@ -0,0 +1,191 @@
+﻿Imports Microsoft.CodeAnalysis.VisualBasic.Syntax.InternalSyntax
+Imports System.Runtime.CompilerServices
+
+Namespace Microsoft.CodeAnalysis.VisualBasic.LanguageFeatures
+
+    Friend Class VisualBasicRequiredLanguageVersionService
+        Private ReadOnly Property _RequiredVersionsForFeature As New Dictionary(Of Feature, VisualBasicRequiredLanguageVersion)()
+        Public Shared ReadOnly Instance As New VisualBasicRequiredLanguageVersionService
+
+        Private Sub New()
+            For Each f As Feature In [Enum].GetValues(GetType(Feature))
+                Dim requiredVersion = New VisualBasicRequiredLanguageVersion(FeatureExtensions.GetLanguageVersion(f))
+                _RequiredVersionsForFeature.Add(f, requiredVersion)
+            Next
+        End Sub
+
+        Public Function GetRequiredLanguageVersion(f As Feature) As VisualBasicRequiredLanguageVersion
+            Return Me._RequiredVersionsForFeature(f)
+        End Function
+    End Class
+
+    Friend Class VisualBasicRequiredLanguageVersion
+        Inherits RequiredLanguageVersion
+
+        Friend ReadOnly Property Version As LanguageVersion
+
+        Friend Sub New(version As LanguageVersion)
+            Me.Version = version
+        End Sub
+
+        Public Overrides Function ToString() As String
+            Return Version.ToDisplayString()
+        End Function
+    End Class
+
+    Module CheckFeatureAvailability
```

Comment:
Need an accessibility modifier

Reply:
I would move this module into a separate .vb file and also remove the `#region` blocks. Class doesn't seem to warrant them.